src/tokens/TokenBalanceItem.test.tsx

```diff
@@ -0,0 +1,104 @@
+import { fireEvent, render } from '@testing-library/react-native'
+import BigNumber from 'bignumber.js'
+import React from 'react'
+import { Provider } from 'react-redux'
+import { AssetsEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { TokenBalance } from 'src/tokens/slice'
+import { TokenBalanceItem } from 'src/tokens/TokenBalanceItem'
+import { createMockStore } from 'test/utils'
+import { mockCusdAddress } from 'test/values'
+
+let mockTokenInfo: TokenBalance
+
+describe('TokenBalanceItem', () => {
+  beforeEach(() => {
+    mockTokenInfo = {
+      balance: new BigNumber('10'),
+      priceUsd: new BigNumber('1'),
+      lastKnownPriceUsd: new BigNumber('1'),
+      symbol: 'cUSD',
+      isCoreToken: true,
+      priceFetchedAt: Date.now(),
+      decimals: 18,
+      name: 'Celo Dollar',
+      imageUrl: '',
+      tokenId: `celo-alfajores:${mockCusdAddress}`,
+      networkId: 'celo-alfajores',
+    } as any
+  })
+
+  it('displays correctly', () => {
+    const { getByText, getByTestId, queryByTestId } = render(
+      <Provider store={createMockStore({})}>
+        <TokenBalanceItem token={mockTokenInfo} />
+      </Provider>
+    )
+
+    expect(getByText('Celo Dollar')).toBeTruthy()
+    expect(getByText('10.00 cUSD')).toBeTruthy()
+    expect(getByText('₱13.30')).toBeTruthy()
+    expect(getByTestId('NetworkLabel')).toBeTruthy()
+    expect(queryByTestId('BridgeLabel')).toBeFalsy()
+  })
+
+  it('displays correctly when token is bridged', () => {
+    mockTokenInfo.bridge = 'Valora Bridge V2'
+    const { getByText, getByTestId } = render(
+      <Provider store={createMockStore({})}>
+        <TokenBalanceItem token={mockTokenInfo} />
+      </Provider>
+    )
+
+    expect(getByText('Celo Dollar')).toBeTruthy()
+    expect(getByText('10.00 cUSD')).toBeTruthy()
+    expect(getByText('₱13.30')).toBeTruthy()
+    expect(getByTestId('BridgeLabel')).toBeTruthy()
+  })
+
+  it('displays correctly when the token network is provided', () => {
+    const { getByText, getByTestId } = render(
+      <Provider store={createMockStore({})}>
+        <TokenBalanceItem token={mockTokenInfo} />
+      </Provider>
+    )
+
+    expect(getByText('Celo Dollar')).toBeTruthy()
+    expect(getByText('10.00 cUSD')).toBeTruthy()
+    expect(getByText('₱13.30')).toBeTruthy()
+    expect(getByTestId('NetworkLabel')).toBeTruthy()
+  })
+
+  it('tracks data about the asset when tapped', () => {
```

Comment:
This test is for actually checking if the component is touchable if onPress is passed and invokes it on clicking right? And not specific to analytics. Could we rename this test?